pandas/util/testing.py

```diff
@@ -829,7 +833,8 @@ def _get_ilevel_values(index, level):
             # get_level_values may change dtype
             _check_types(left.levels[level], right.levels[level], obj=obj)
 
-    if check_exact:
+    run_exact_index_check = check_exact and check_categorical
```

Comment:
you don't need a variable, just do the:

``if checkexact and check_categorical:``